rclpy/src/rclpy/_rclpy.c

```diff
@@ -155,14 +155,51 @@ rclpy_trigger_guard_condition(PyObject * Py_UNUSED(self), PyObject * args)
  * Raises RuntimeError if rcl could not be initialized
  */
 static PyObject *
-rclpy_init(PyObject * Py_UNUSED(self), PyObject * Py_UNUSED(args))
+rclpy_init(PyObject * Py_UNUSED(self), PyObject * args)
 {
-  // TODO(esteve): parse args
-  rcl_ret_t ret = rcl_init(0, NULL, rcl_get_default_allocator());
-  if (ret != RCL_RET_OK) {
-    PyErr_Format(PyExc_RuntimeError,
-      "Failed to init: %s", rcl_get_error_string_safe());
-    rcl_reset_error();
+  // Expect one argument which is a list of strings
+  PyObject * pyargs;
+  if (!PyArg_ParseTuple(args, "O", &pyargs)) {
+    return NULL;
```

Comment:
Just to check, these functions setup the exceptions right? That's why only returning null here is fine?

Reply:
Yup, [PyArg_ParseTuple](https://docs.python.org/3.5/c-api/arg.html#c.PyArg_ParseTuple) returns false and sets an exception on failure.

Reply:
Added comments about exceptions already being set in 023ccafde1fdf0026b47f19a06d978f19f946dcc